app/controllers/api/migrations.php

```diff
@@ -290,6 +300,136 @@
+                'migrations',
+                [
+                    Query::notEqual('stage', 'finished'),
+                    Query::equal('resourceId', [$resourceId]),
+                ]
+            );
+
+            return !$exists->isEmpty();
+        });
+
+        if ($isInProgress || (!$isAPIKey && !$isPrivilegedUser)) {
+            throw new Exception(Exception::MIGRATION_IN_PROGRESS, 'An import is already in progress for this collection.');
+        }
+
+        $bucket = Authorization::skip(fn () => $dbForProject->getDocument('buckets', $bucketId));
+
+        if ($bucket->isEmpty() || (!$isAPIKey && !$isPrivilegedUser)) {
+            throw new Exception(Exception::STORAGE_BUCKET_NOT_FOUND);
+        }
+
+        $file = Authorization::skip(fn () => $dbForProject->getDocument('bucket_' . $bucket->getInternalId(), $fileId));
+        if ($file->isEmpty()) {
+            throw new Exception(Exception::STORAGE_FILE_NOT_FOUND);
+        }
+
+        $path = $file->getAttribute('path', '');
+        if (!$deviceForFiles->exists($path)) {
+            throw new Exception(Exception::STORAGE_FILE_NOT_FOUND, 'File not found in ' . $path);
+        }
+
+        // read file content.
+        $source = $deviceForFiles->read($path);
+
+        // decrypt
+        if (!empty($file->getAttribute('openSSLCipher'))) {
+            $source = OpenSSL::decrypt(
+                $source,
+                $file->getAttribute('openSSLCipher'),
+                System::getEnv('_APP_OPENSSL_KEY_V' . $file->getAttribute('openSSLVersion')),
+                0,
+                \hex2bin($file->getAttribute('openSSLIV')),
+                \hex2bin($file->getAttribute('openSSLTag'))
+            );
+        }
+
+        // decompress
+        switch ($file->getAttribute('algorithm', Compression::NONE)) {
+            case Compression::ZSTD:
+                $compressor = new Zstd();
+                $source = $compressor->decompress($source);
+                break;
+            case Compression::GZIP:
+                $compressor = new GZIP();
+                $source = $compressor->decompress($source);
+                break;
+        }
+
+        // copy to temporary folder
+        $migrationId = ID::unique();
+        $path = $deviceForCsvImports->getRoot() . '/' . $migrationId . '_' . $fileId . '.csv';
+        $deviceForCsvImports->write($path, $source, 'text/csv');
+        $fileSize = $deviceForCsvImports->getFileSize($path);
+        $resources = Transfer::extractServices([Transfer::GROUP_DATABASES]);
+
+        $migration = $dbForProject->createDocument('migrations', new Document([
+            '$id' => $migrationId,
+            'status' => 'pending',
+            'stage' => 'init',
+            'source' => Csv::getName(),
+            'destination' => Appwrite::class::getName(),
+            'resources' => $resources,
+            'resourceId' => $resourceId,
+            'resourceType' => Resource::TYPE_DATABASE,
```

Comment:
I think this needs be collection, since it can only be one per file

Reply:
I think it'd throw an error - `Resource type must be one of bucket, database, function, user, team`. Only the top layer of the services is expected here.